mne/tests/test_epochs.py

```diff
@@ -57,6 +59,71 @@
+                       [3, 0, 2], [3, 0, 1],
+                       [5, 0, 2], [5, 0, 1], [5, 0, 3],
+                       [7, 0, 1]])
+
+    with pytest.raises(RuntimeError, match='Event time samples were not uniq'):
+        _handle_duplicate_events(EVENTS, EVENT_ID, event_repeated='error')
+
+    with pytest.raises(ValueError, match='`event_repeated` must be one of '):
+        _handle_duplicate_events(EVENTS, EVENT_ID, event_repeated='bogus')
+
+    events, event_id = _handle_duplicate_events(EVENTS, EVENT_ID, 'drop')
+    assert_array_equal(events, [[0, 0, 1], [3, 0, 2], [5, 0, 2], [7, 0, 1]])
+    assert event_id == {'aud': 1, 'vis': 2}
+
+    events, event_id = _handle_duplicate_events(EVENTS, EVENT_ID, 'merge')
+    assert_array_equal(events[0][-1], events[1][-1])
+    assert_array_equal(events, [[0, 0, 4], [3, 0, 4], [5, 0, 5], [7, 0, 1]])
+    assert 'aud/vis' in event_id.keys()
+    assert set(event_id.keys()) == set(['aud', 'aud/vis', 'aud/foo/vis'])
+    assert event_id['aud/vis'] == 4
+
+    # Test early return with no changes: no error for wrong event_repeated arg
+    fine_events = np.array([[0, 0, 1], [1, 0, 2]])
+    events, event_id = _handle_duplicate_events(fine_events, EVENT_ID, 'no')
+    assert event_id == EVENT_ID
+    np.testing.assert_array_equal(events, fine_events)
+    del fine_events
+
+    # Test falling back on 0 for heterogeneous "prior-to-event" codes
+    # order of third column does not determine new event_id key, we always
+    # take components, sort, and join on "/"
+    # should make new event_id value: 5 (because 1,2,3,4 are taken)
+    heterogeneous_events = np.array([[0, 3, 2], [0, 4, 1]])
+    events, event_id = _handle_duplicate_events(heterogeneous_events,
+                                                EVENT_ID, 'merge')
+    assert 'aud/vis' in event_id.keys()
+    assert set(event_id.keys()) == set(['aud/vis'])
+    assert event_id['aud/vis'] == 5
+    np.testing.assert_array_equal(events, np.array([[0, 0, 5], ]))
+    del heterogeneous_events
+
+    # Test keeping a homogeneous "prior-to-event" code
+    homogeneous_events = np.array([[0, 99, 1], [0, 99, 2],
+                                   [1, 0, 1], [2, 0, 2]])
+    events, event_id = _handle_duplicate_events(homogeneous_events,
+                                                EVENT_ID, 'merge')
+    assert set(event_id.keys()) == set(['aud', 'vis', 'aud/vis'])
+    np.testing.assert_array_equal(events, np.array([[0, 99, 4],
+                                                    [1, 0, 1], [2, 0, 2]]))
+    del homogeneous_events
+
+    # Test dropping instead of merging, if event_codes to be merged are equal
+    equal_events = np.array([[0, 0, 1], [0, 0, 1]])
+    events, event_id = _handle_duplicate_events(equal_events,
+                                                EVENT_ID, 'merge')
+    np.testing.assert_array_equal(events, np.array([[0, 0, 1], ]))
+    assert set(event_id.keys()) == set(['aud'])
+    del equal_events
```

Comment:
```suggestion
```

If the function terminates here, why del anything?

Reply:
yes, that line can go ...

I was deleting the variables before so that the output of pytest addons is nicer. I don't know which one it is, but it prints your local variables whenever you hit an error --> and I don't want this to be cluttered with irrelevant vars from prior tests